### PR TITLE
Post on asynchronous JS

### PR DESCRIPTION
**Pull request by iamleeg**

### CODE REVIEW

File: content/tech_post/async-js.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+---
+title: "If not now, then when? Asynchronous JavaScript"
+date: 2021-08-17T11:25:53+01:00
+draft: false
+---
+
+We use Javascript and its type-safe sibling, TypeScript, on a lot of projects. For example, in our work on [Global.health: a Data Science Initiative](https://global.health), we use TypeScript both for the web application and for the API backing it, via the node.js engine. TypeScript gets rewritten into JavaScript before it is run, so I'll just use JavaScript from now on without worrying about how we got that JavaScript.
```
Comment (martinjrobins) [inline content/tech_post/async-js.md:7]:
```suggestion
We use JavaScript and its type-safe sibling, TypeScript, on a lot of projects. For example, in our work on [Global.health: a Data Science Initiative](https://global.health), we use TypeScript both for the web application and for the API backing it, via the node.js engine. TypeScript gets rewritten into JavaScript before it is run, so to keep things as simple as possible I'll just use JavaScript from now on.
```

File: content/tech_post/async-js.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+Underlying a JavaScript engine, such as the [v8](https://v8.dev) project that powers both Chrome and node.js, is an _event loop_. This can be thought of as a queue of things that need doing, and a program that asks "is there anything I still need to do?". If there is something to do, the event loop runs the appropriate JS. If not, it either waits until new work arrives, or exits depending on the context.
+
+If you have some long-running task, you can have it do its work in the background and add an event to the queue when it's done for the event loop to pick up and run. Now the question is how do we do that from JS, and the answer is that there are many ways.
+
+## Callback Functions
+
+The first solution provided to scheduling work on the event loop was the callback. You would invoke your long-running task, giving it a function to call when it was finished. The function got access to the result of the task, so you could deal with errors or process the results. It looks like this:
```
Comment (martinjrobins) [inline content/tech_post/async-js.md:19]:
```suggestion
Historically, the first solution provided to scheduling work on the event loop was the callback. You would invoke your long-running task, giving it a function to call when it was finished. The function got access to the result of the task, so you could deal with errors or process the results. Using as an example a database query as our long-running task, a callback would look like this:
```

**Review by martinjrobins — APPROVED** [review]
thanks @iamleeg! A great read, I've made a few minor suggestions

**Review by abhidg — APPROVED** [review]
This post is full of promise.

**Pull request merged**